src/core/qgsmapthemecollection.cpp

```diff
@@ -276,6 +276,19 @@ void QgsMapThemeCollection::update( const QString &name, const MapThemeRecord &s
   emit mapThemesChanged();
 }
 
+
+void QgsMapThemeCollection::renameMapTheme( const QString &name,  const QString &newName ) //, const MapThemeRecord &state )
+{
+  if ( !mMapThemes.contains( name ) || mMapThemes.contains( newName ) )
+    return;
```

Comment:
I'd make this function return a bool, and return false here

Reply:
Done but I'm not sure of what it implies nor whether what i did is what you expected

Reply:
Looks good!